src/supla/sensor/SGP41.h

```diff
@@ -0,0 +1,142 @@
+/*
+ Copyright (C) malarz
+
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ as published by the Free Software Foundation; either version 2
+ of the License, or (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+*/
+
+#ifndef SRC_SUPLA_SENSOR_SGP41_H_
+#define SRC_SUPLA_SENSOR_SGP41_H_
+
+// Dependency: Sensirion I2C SGP41 Arduino Library
+// - use library manager to install it
+// https://github.com/Sensirion/arduino-i2c-sgp41
+
+#include "SensirionI2CSgp41.h"
+
+#include <supla/sensor/therm_hygro_meter.h>
+#include <supla/sensor/general_purpose_measurement.h>
+
+namespace Supla {
+namespace Sensor {
+class SGP41 : public Element {
+ public:
+  explicit SGP41(ThermHygroMeter *ptr = nullptr) {
+    th = ptr;
+    vocchannel = new GeneralPurposeMeasurement();
+    vocchannel->setDefaultUnitAfterValue("");
+    vocchannel->setInitialCaption("VOC index");
+    vocchannel->getChannel()->setDefaultIcon(8);
+    vocchannel->setDefaultValuePrecision(1);
+
+    noxchannel = new GeneralPurposeMeasurement();
+    noxchannel->setDefaultUnitAfterValue("");
+    noxchannel->setInitialCaption("NOx index");
+    noxchannel->getChannel()->setDefaultIcon(8);
+    noxchannel->setDefaultValuePrecision(1);
+  }
+
+  double getVOC() {
+    return voc;
+  }
+
+  double getNOx() {
+    return nox;
+  }
+
```

Comment:
not needed?
You can always do: sgp->getVOCchannel()->getValue(); (or getCalculatedValue())